### PR TITLE
Fix title from draft being dismissed (#718)

### PR DESCRIPTION
**Pull request by jp30566347**
Add a way to bypass the title generation query when the url change is from local storage (draft) and not user interaction.

## Summary by CodeRabbit

- **New Features**
	- Enhanced form input handling to recognize draft changes, improving the user experience for data auto-saved from local storage.
- **Refactor**
	- Improved logic in link form validation to accurately detect empty titles, especially for draft changes, ensuring more reliable form submissions.

### CODE REVIEW

**Comment by jp30566347** [comment]
Suggestion fix for #718

**Review by ekzyis — CHANGES_REQUESTED** [review]
This fixes that the draft value is overridden:

https://github.com/stackernews/stacker.news/assets/27162016/254c0217-a04a-4011-b1f8-c967f3563ab8

But it breaks fetching the title on page load:

https://github.com/stackernews/stacker.news/assets/27162016/16c50a35-3415-4705-ada3-93040f1cbb5e

The title is still fetched if the URL is changed after page load.

**Comment by jp30566347** [comment]
Oh, I missed that requirement, I will take a closer look!

**Comment by jp30566347** [comment]
Updated fix, I'm open to suggestions because it's a bit smelly.
The difficulty here is getting the draft value of the title from the onChange event of the url field.
Note that if there is no draft for a specific field, the onChange is not fired, so it's impossible from the parent component to determine if a draft exists or not. 

I was also wondering if we should regenerate a title (override existing one) if the url is changed? To me it sounds like it but right now it's not.

**Comment by ekzyis** [comment]
> I was also wondering if we should regenerate a title (override existing one) if the url is changed? To me it sounds like it but right now it's not.

I agree with you. If you change the URL, the old title will most likely no longer make sense. So I think regenerating the title via fetch would make sense.

However, unless it's a regression (like the page load fetch), you don't need to care about it in this PR. Currently not deep in that part of the code right now so not sure if that is already a thing, will try to review the changes later today.

But if the code gets too messy, maybe we can relax requirements a bit. Never overriding draft values might be more important. Not sure yet though.

File: components/form.js [hunk 1]
```diff
@@ -453,7 +453,8 @@ function InputInner ({
         const isNumeric = /^[0-9]+$/.test(draft)
         const numericExpected = typeof field.value === 'number'
         helpers.setValue(isNumeric && numericExpected ? parseInt(draft) : draft)
-        onChange && onChange(formik, { target: { value: draft } })
+        // draft means change event is from local storage
+        onChange && onChange(formik, { target: { value: draft, draft: true } })
```
Comment (huumn) [inline components/form.js:457]: It doesn't look like we are using `draft: true` anymore, are we?

Comment (jp30566347) [reply components/form.js]: I removed it. Good catch.

**Comment by huumn** [comment]
I agree it's smelly but it's smelly in an obvious way at least. If this is something we want, it's either this or a new abstraction which this change doesn't justify alone.

I don't think we have this kind of field interdependence anywhere else.

**Pull request merged**